### DISCUSSION TITLE
Permanent downloadUrl

### DISCUSSION THREAD

**Discussion by juanantoniosantanamedina** [supabase/supabase#1708] [Feature Requests]
At the moment in order to display an image stored in the storage we need to get a downloadUrl that expires in a certain period of time.

That sounds good , but there are certain scenarios where I would prefer to store that url in the database and use it any time I need it. A good example for that is a list of posts where the avatar of the poster user is shown. Do I really want to download that avatar in each element for the list? I just need to have the url to the image in the profiles table and read it.

Maybe this is already possible and I don't know how to do it !!

Any help please?

**Comment by silentworks** [comment]
The Supabase team are currently working on a feature to allow you to set certain buckets to be public, so that way you don't need to get the downloadUrl.

**Reply by juanantoniosantanamedina** [reply]
Thanks, that will be great. Any idea of when it will be ready?

**Reply by inian** [reply]
For anyone coming across this, we have [launched public buckets](https://supabase.io/docs/guides/storage#accessing-objects) now. You should be able to access assets in a public bucket without a token or a Authorization header.
